specs/grafana_event_handling_in_monitoring_integration.adoc

```diff
@@ -0,0 +1,313 @@
+
+Alert from API request: http://{ip}:3000/api/alerts/6
+{
+Id: 6,
+Version: 0,
+OrgId: 1,
+DashboardId: 7,
+PanelId: 3,
+Name: "Panel Title alert",
+Message: "alert",
+Severity: "",
+State: "alerting",
+Handler: 1,
+Silenced: false,
+ExecutionError: " ",
+Frequency: 5,
+EvalData: - {
+evalMatches: - [
+- {
+metric: "collectd.dhcp42-170_lab_eng_blr_redhat_com.memory.memory-free",
+tags: null,
+value: 6961614848
+}
+]
+},
+NewStateDate: "2017-07-18T23:07:16+05:30",
+StateChanges: 291,
+Created: "2017-09-18T14:36:37+05:30",
+Updated: "2017-07-19T04:37:11+05:30",
+Settings: - {
+conditions: - [
+- {
+evaluator: - {
+params: - [
+6956963696
+],
+type: "gt"
+},
+operator: - {
+type: "and"
+},
+query: - {
+datasourceId: 3,
+model: - {
+refId: "A",
+target: "collectd.dhcp42-170_lab_eng_blr_redhat_com.memory.memory-free"
+},
+params: - [
+"A",
+"5m",
+"now"
+]
+},
+reducer: - {
+params:[],
+type: "avg"
+},
+type: "query"
+}
+],
+executionErrorState: "alerting",
+frequency: "5s",
+handler: 1,
+message: "alert",
+name: "Panel Title alert",
+noDataState: "alerting",
+notifications:[]
+}
+}
+    
+```
+
+After receiving alert then it will converted to current tendrl alert structure. 
+```
+Alert_id
+node_id
+time_stamp
+resource
+current_value
+tags
+alert_type
+severity
+significance
+ackedby
+acked
+ack_comment
+acked_at
+pid
+source
+```
+
+Once all the required details of alert event (grafana specific data and tendrl specific metadata)
+is collected from the respective data sources(grafana and tendrl objects) it 
+is supposed to be notified to the user which will be done by the Notifier component.
+
+Also the alerts are to be stored in etcd for logging purposes.This will be
+done by the node-agent.
+
+The Notifier can receive the Event in 2 ways:
+
+* The alert events are stored in etcd. The notifier could collect the alerts
+  from etcd and then using the configuration file in etcd it could notify the 
+  respective user.
+  
+  Drawback:
+  
+  ** Pooling etcd for new alerts is not efficient.
+  
+* The node-agent can directly pass the alert event to the notifier via an endpoint
+  exposed by the notifier.
+
+  Advantage:
+  
+  ** Avoids pooling etcd for new alerts.
+
+The Notifier will notify the user of the events via notification channels
+like SMTP.
+
+=== Alternatives:
+
+None
+
+
+=== Data model impact:
+
+ No changes in existing structure.
+
+
+=== Impacted Modules:
+
+==== Tendrl API impact:
+
+None
+
+==== Notifications/Monitoring impact:
+
+None
+
+==== Tendrl/common impact:
+None
+
+==== Tendrl/node_agent impact:
+
+None
+
+==== Sds integration impact:
+
+None
+
+==== Tendrl/performance-monitoring impact:
+
+A  new component "Monitoring Integration" will be introduced to
```

Comment:
I guess a lot of functionality in perf-mon is not needed, so instead of using the perf-mon we are going to create a new component (a separate repo).
i will double check with Mrugesh